docs/mvvm/MigratingFromMvvmLight.md

```diff
@@ -0,0 +1,842 @@
+
+```csharp
+// MvvmLight
+this.VerifyPropertyName(nameof(this.MyProperty));
+
+// Toolkit.Mvvm
+// No direct replacement, remove
+```
+
+### ObservableObject Properties
+
+#### PropertyChangedHandler
+
+`PropertyChangedHandler` does not have a direct replacement.
+
+To raise a property changed event via the `PropertyChanged` event handler, you need to call the `OnPropertyChanged` method instead. 
+
+```csharp
+// MvvmLight
+PropertyChangedEventHandler handler = this.PropertyChangedHandler;
+
+// Toolkit.Mvvm
+this.OnPropertyChanged(nameof(this.MyProperty));
+```
+
+## Migrating ViewModelBase
+
+The following steps focus on migrating your existing components which take advantage of the `ViewModelBase` of the MvvmLight Toolkit. 
+
+The Windows Community Toolkit MVVM framework provides an [`ObservableRecipient`](ObservableRecipient) type that provides similar functionality. 
+
+Below are a list of migrations that will need to be performed if being used in your current solution.
+
+### ViewModelBase Methods
+
+#### Set<T> ( string, ref T, T, bool )
+
+`Set<T>(string, ref T, T, bool)` does not have a like-for-like method signature replacement. 
+
+However, `SetProperty<T>(ref T, T, bool, string)` provides the same functionality with re-ordered parameters.
+
+```csharp
+// MvvmLight
+this.Set(nameof(this.MyProperty), ref this.myProperty, value, true);
+
+// Toolkit.Mvvm
+this.SetProperty(ref this.myProperty, value, true, nameof(this.MyProperty));
+```
+
+Note, the value and broadcast boolean parameters are not optional in the Toolkit's implementation and must be provided to use this method. The reason for this change is that by omitting the broadcast parameter when calling this method, it will by default call the ObservableObject's `SetProperty` method.
+
+Also, string parameter is not required if the method is being called from the property's setter as it is inferred from the caller member name.
+
+#### Set<T> ( ref T, T, bool, string )
+
+`Set<T>(ref T, T, bool, string)` has a renamed direct replacement, `SetProperty<T>(ref T, T, bool, string)`. 
+
+```csharp
+// MvvmLight
+this.Set(ref this.myProperty, value, true, nameof(this.MyProperty));
+
+// Toolkit.Mvvm
+this.SetProperty(ref this.myProperty, value, true, nameof(this.MyProperty));
+```
+
+Note, string parameter is not required if the method is being called from the property's setter as it is inferred from the caller member name.
+
+#### Set<T> ( Expression, ref T, T, bool )
+
+`Set<T>(Expression, ref T, T, bool)` does not have a direct replacement. 
+
+It is recommended for improved performance that you replace this with the Toolkit's `SetProperty<T>(ref T, T, bool, string)` using the `nameof` keyword instead.
+
+```csharp
+// MvvmLight
+this.Set<MyObject>(() => this.MyProperty, ref this.myProperty, value, true);
+
+// Toolkit.Mvvm
+this.SetProperty<MyObject>(ref this.myProperty, value, true, nameof(this.MyProperty));
```

Comment:
Why the explicit type argument here? The C# compiler should be able to infer them in this case, no? 🤔